### ISSUE TITLE
Solar Clipping on Givenergy Hybrid Inverter

### ISSUE THREAD

**Issue by robyholmes** [springfall2008/batpred#1250]
**Describe the bug**
Predbat isn't trying to stop hybrid inverters from clipping by discharging the battery before the peak of the day. So for example a 5kw inverter with the battery at 100%, and solar over forecast over the 5kw is not being discharged in the morning to avoid it clipping.

**Expected behavior**
Planned discharging in the morning when the solar forecast is less than the inverter capacity, creating room in the battery so DC to DC charging.

**Predbat version**

V8.1.1 & Main on 24/06/2024
Predbat Add-on
None AppDaemon

**Environment details**

Givenergy Hybrid 5kw Gen 3
2x 9.5kwh Batteries
7.2kwh Peak Solar Panels
HAOS 2024.6.4

**Screenshots**
Taken from Facebook Post (24/06/2024)
V8.1.2
![448924556_10161967827264744_7476904610614463826_n](https://github.com/springfall2008/batpred/assets/28155240/b616863e-703c-46da-ab3d-9e9a7600675b)

**Log file**
To follow soon, as I missed downloading it today before the peak today. Tomorrow doesn't look to be over the peak so might be later this week.

**Comment by gcoan** [comment]
There are a few other issues already raised related to inverter clipping and Predbat's management of it, #886 and #1206 as examples.

At the moment Predbat doesn't yet have actively try to reduce clipping, but this is something Trefor is working on.

There are different strategies such as force discharging the battery as you said, or letting the battery trickle charge at a lower rate so it doesn't fill up too early in the day.

To prevent duplication of issues can you see if you can add your experience/feedback to one of the existing issues please

**Comment by robyholmes** [comment]
I understand, however as Trefor was talking to me and others about this issue on facebook, trying different settings etc. I thought putting it on Github as an issue was the best way of tracking it, rather than facebook replies.

**Comment by gcoan** [comment]
Agreed it's a hot topic that others have been raising for some time.  I was just suggesting that you add to an existing issue rather than duplicate the conversation about how to implement clipping control over multiple tickets so you join an existing conversation rather than starting a new one.

Have a look at the other issues though

**Comment by robyholmes** [comment]
So far today, things are looking food. It looks to be trying to avoid clipping by putting some planned discharges in.
![image](https://github.com/springfall2008/batpred/assets/28155240/2b51d696-f5c7-41fb-8908-a6b83d070d01)

**Comment by springfall2008** [comment]
As this ticket is old I'm going to close it, please re-open if the issue is there on latest.
